doc/api/esm.md

```diff
@@ -1008,7 +1002,7 @@ _isImports_, _conditions_)
 > 5. Let _packageURL_ be the result of **LOOKUP\_PACKAGE\_SCOPE**(_url_).
 > 6. Let _pjson_ be the result of **READ\_PACKAGE\_JSON**(_packageURL_).
 > 7. If _pjson?.type_ exists and is _"module"_, then
->    1. If _url_ ends in _".js"_, then
+>    1. If _url_ ends in _".js"_ or lacks file extension, then
```

Comment:
type module controls .js, but there’s ways to use ESM without type module - it shouldn’t be overloaded to also control extensionless, and there needs to be a way to use extensionless ESM without type module imo.

Reply:
Please elaborate if you mean a need to reword this or suggest changes to the implementation.
The `type` controls how any JS code is interpreted. The exceptions are files with explicitly typed extensions (`.cjs`, `.mjs`, maybe future formats) and with unknown extensions (these should throw to ensure that it breakage is minimized if we add future formats).

Reply:
I’m suggesting an additional package.json key to control extensionless. “type” only controls how .js is interpreted, not “any” js code.